connection/connection.go

```diff
@@ -65,12 +69,32 @@ type Option func(o *options)
 // connection got lost. If that callback returns true, the connection
 // is restablished. Otherwise the connection is left as it is and
 // all future gRPC calls using it will fail with status.Unavailable.
+// Only one of ExitOnConnectionLoss and OnConnectionLoss can be used.
 func OnConnectionLoss(reconnect func() bool) Option {
 	return func(o *options) {
 		o.reconnect = reconnect
 	}
 }
 
+// ExitOnConnectionLoss exits when connection gets lost. Optionally it
+// writes error to given file. Use with /dev/termination-log to get a nice
+// error in Kubernetes.
+// Only one of ExitOnConnectionLoss and OnConnectionLoss can be used.
+func ExitOnConnectionLoss(terminationLogPath string) Option {
+	return func(o *options) {
+		o.reconnect = func() bool {
+			terminationMsg := "Lost connection to CSI driver, exiting"
+			if terminationLogPath != "" {
+				if err := ioutil.WriteFile(terminationLogPath, []byte(terminationMsg), 0644); err != nil {
+					klog.Errorf("%s: %s", terminationLogPath, err)
+				}
+			}
+			klog.Fatalf(terminationMsg)
```

Comment:
Won't that raise a panic? That might get caught further up in the call chain and even if it isn't.

What is the output? Does it include a stack backtrace? I don't think we want that - just the message and the exit the process.

Reply:
It just prints the message, no panic.

```
I0214 16:59:44.965050   26907 shared_informer.go:123] caches populated
E0214 16:59:46.528572   26907 connection.go:127] Lost connection to unix:///tmp/csi.sock.
E0214 16:59:46.528679   26907 connection.go:85] /dev/termination-log: open /dev/termination-log: permission denied
F0214 16:59:46.528735   26907 connection.go:88] Lost connection to CSI driver, exiting

$ echo $?
255
```

Reply:
That's okay then. I was thinking of `klog.Panicf`, my bad.